code/game/objects/structures/crates_lockers/closets/secure/guncabinet.dm

```diff
@@ -1,51 +1,50 @@
-/obj/structure/closet/secure_closet/guncabinet
-	name = "gun cabinet"
-	icon = 'icons/obj/guncabinet.dmi'
-	icon_state = "base"
-	req_one_access = list(access_armory)
-	closet_appearance = null
-
-/obj/structure/closet/secure_closet/guncabinet/Initialize()
-	. = ..()
-	update_icon()
-
-/obj/structure/closet/secure_closet/guncabinet/toggle()
-	..()
-	update_icon()
-
-/obj/structure/closet/secure_closet/guncabinet/update_icon()
-	cut_overlays()
-	if(opened)
-		add_overlay("door_open")
-	else
-		var/lazors = 0
-		var/shottas = 0
-		for (var/obj/item/gun/G in contents)
-			if (istype(G, /obj/item/gun/energy))
-				lazors++
-			if (istype(G, /obj/item/gun/projectile))
-				shottas++
-		for (var/i = 0 to 2)
-			if(lazors || shottas) // only make icons if we have one of the two types.
-				var/image/gun = image(icon(src.icon))
-				if (lazors > shottas)
-					lazors--
-					gun.icon_state = "laser"
-				else if (shottas)
-					shottas--
-					gun.icon_state = "projectile"
-				gun.pixel_x = i*4
-				add_overlay(gun)
-
-		add_overlay("door")
-
-		if(sealed)
-			add_overlay("sealed")
-
-		if(broken)
-			add_overlay("broken")
-		else if (locked)
-			add_overlay("locked")
-		else
-			add_overlay("open")
-
+/obj/structure/closet/secure_closet/guncabinet
+	name = "gun cabinet"
+	icon = 'icons/obj/guncabinet.dmi'
+	icon_state = "base"
+	req_one_access = list(access_armory)
+	closet_appearance = null
+
+/obj/structure/closet/secure_closet/guncabinet/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/structure/closet/secure_closet/guncabinet/toggle()
+	..()
+	update_icon()
+
+/obj/structure/closet/secure_closet/guncabinet/update_icon()
+	cut_overlays()
+	var/list/add = list()

```

Comment:
replaces doing a bunch of individual adds and queues with building a list first and then passing it all at once